src/backend/cdb/dispatcher/cdbdisp_async.c

```diff
@@ -914,38 +989,41 @@ static void
 signalQEs(CdbDispatchCmdAsync *pParms)
 {
 	int			i;
-	DispatchWaitMode waitMode = pParms->waitMode;
 
 	for (i = 0; i < pParms->dispatchCount; i++)
-	{
-		char		errbuf[256];
-		bool		sent = false;
-		CdbDispatchResult *dispatchResult = pParms->dispatchResultPtrArray[i];
+		signalQE(pParms->dispatchResultPtrArray[i], pParms->waitMode);
+}
 
-		Assert(dispatchResult != NULL);
-		SegmentDatabaseDescriptor *segdbDesc = dispatchResult->segdbDesc;
+/*
+ * Send finish or cancel signal to QE if needed.
+ */
+static void
+signalQE(CdbDispatchResult *dispatchResult, DispatchWaitMode waitMode)
```

Comment:
What is the point now of refactoring the `signalQEs` function and adding a new `signalQE` function?